### PR TITLE
Add DomainID and DomainName options to apiserver

### PR DESCRIPTION
**Pull request by mkumatag**
Fix for #25066

---

This change is [Reviewable](https://reviewable.io/reviews/kubernetes/kubernetes/25259)

### CODE REVIEW

**Comment by kfox1111** [comment]
I have a patch that implements:
https://github.com/kubernetes/kubernetes/issues/25066
in the works that also fixes this I think. I think we should wait on this patch.

This patch here will only work if you only have one domain at your site. Which is not always true. If you are using Heat, then its guaranteed that Heat couldn't integrate with k8s in this mode since it uses a second domain for all of its trust users.

**Comment by mkumatag** [comment]
@kfox1111 can you please point me the patch are you talking about. I'm interested in knowing how are you calling gophercloud with multiple domainID.

**Comment by kfox1111** [comment]
WorkIng through the cla. Then I can post it. Hopefully soon. Until then, most of the details of how it works are talked about in the issue.

Short answer, domain ids are needed at username/password resolve time. Thats better handled on the client side and keystone tokens passed from the client to the apiserver and tokens are validated at the server. No domain id is needed on the server for tokens since they are baked into the token.

**Comment by erictune** [comment]
@kfox1111 how does this relate to #25624

**Comment by liggitt** [comment]
when api-server flags migrate to a config file in the future, do we want config files referencing config files?

**Comment by mkumatag** [comment]
I agree with @kfox1111 , option 2 is better solution where your have config file contains all the options except username and password.

**Comment by kfox1111** [comment]
@liggitt there are other flags using config files too, such as cinder volumes/neutron lbaas that option 2 code would share code with, could be updated all at once to pull from a shared config with different [section]'s. or a yaml or something if that is the chosen format.

File: cmd/kube-apiserver/app/options/options.go [hunk 1]
```diff
@@ -169,6 +171,8 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ServiceAccountKeyFile, "service-account-key-file", s.ServiceAccountKeyFile, "File containing PEM-encoded x509 RSA private or public key, used to verify ServiceAccount tokens. If unspecified, --tls-private-key-file is used.")
 	fs.BoolVar(&s.ServiceAccountLookup, "service-account-lookup", s.ServiceAccountLookup, "If true, validate ServiceAccount tokens exist in etcd as part of authentication.")
 	fs.StringVar(&s.KeystoneURL, "experimental-keystone-url", s.KeystoneURL, "If passed, activates the keystone authentication plugin")
+        fs.StringVar(&s.KeystoneDomainID, "experimental-keystone-domainid", s.KeystoneDomainID, "DomainID for keystone authentication")
```
Comment (linhuacheng) [inline cmd/kube-apiserver/app/options/options.go]: for naming consistency, should this be named as experimental-keystone-domain-id and experimental-keystone-domain-name instead?

Comment (mkumatag) [inline cmd/kube-apiserver/app/options/options.go]: All these options may go away, waiting for #25624 to be merged. Need more work here after that PR merged.

**Comment by erictune** [comment]
ok-to-test

**Pull request closed without merging**